src/ca/caContext.h

```diff
@@ -0,0 +1,51 @@
+/**
+ * Copyright - See the COPYRIGHT that is included with this distribution.
+ * pvAccessCPP is distributed subject to a Software License Agreement found
+ * in file LICENSE that is included with this distribution.
+ */
+
+#ifndef INC_caContext_H
+#define INC_caContext_H
+
+#include <cadef.h>
+#include <pv/pvAccess.h>
+
+namespace epics {
+namespace pvAccess {
+namespace ca {
+
+
+class Attach;
+class CAContext;
+typedef std::tr1::shared_ptr<CAContext> CAContextPtr;
+
+class CAContext
+{
+public:
+    CAContext();
+    ~CAContext();
+private:
+    ca_client_context* ca_context;
+
+private:    // Internal API
+    friend class Attach;
+    ca_client_context* attach();
+    void detach(ca_client_context* restore);
+};
+
+class Attach
+{
+public:
+    Attach(const CAContextPtr & to) :
+        context(to), saved_context(to->attach()) {}
+    ~Attach() {
+        context->detach(saved_context);
+    }
+private:
+    CAContextPtr context;
```

Comment:
Make this a CAContext &

Reply:
This works, @mdavidsaver I assume this is what you meant:
```patch
diff --git a/src/ca/caContext.h b/src/ca/caContext.h
index 620cb20f..6830c847 100644
--- a/src/ca/caContext.h
+++ b/src/ca/caContext.h
@@ -37,12 +37,12 @@ class Attach
 {
 public:
     Attach(const CAContextPtr & to) :
-        context(to), saved_context(to->attach()) {}
+        context(*to), saved_context(context.attach()) {}
     ~Attach() {
-        context->detach(saved_context);
+        context.detach(saved_context);
     }
 private:
-    CAContextPtr context;
+    CAContext & context;
     ca_client_context* saved_context;
 };
 
```

Reply:
Yes.

Also, as a matter of habit, I start by marking any single argument constructor with the [`explicit`](https://en.cppreference.com/w/cpp/language/explicit) specifier to inhibit some implicit casts.  I rarely intend to allow sure of such casts, and this is a way of preventing surprise.

Reply:
`explicit` added to the constructor, thanks.